dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,147 @@
+commonly used approaches.
+
+The recommendation is:
+
+- If you have HTTP/HTTPS -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIP], or
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePort].
+
+
+The admin configures DNS or wildcard DNS records to point to IP addresses
+in the cluster that are exposed outside the cluster.  These addresses are
+commonly the IP addresses of nodes, Virtual IP addresses (VRRP) or Service
+NodePort addreses, Service externalIPs, Service ingressIP, etc..
+
+The admin must make sure the local firewall permits the request to reach
+the IP address.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A router is configured
+to accept external reuqests and proxy them based on the configured routes.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+An administrator can create a wild-card DNS entry, and then set up a router,
+after that the users can self-service the ingress without having to bother
+the admins. The router has controls to allow the admin to specify whether
+the users can self-provision hostnames, or if they must fit a pattern the
+admin defines. The other solutions require the admin to do it, or it requires
+that they delegate a lot of privilege.
+
+The admin can create multiple routers and set up router
+xref:../architecture/core_concepts/routes.adoc:#router-sharding[shards]
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
+from the set of routes.
+
+The router IP can be the node's IP or a VRRP provided VIP
+(xref:../admin_guide/high_availability.adoc#[virtual IP]) or with a range TCP
+xref:http://kubernetes.io/docs/user-guide/services/#type-loadbalancer[load balancer]
+(cloud,
+xref:../architecture/core_concepts/routes.adoc:#f5-router[F5], ingerssIP)
+as the destination IP.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to configure a router for high availbility service using
+multiple replicas.
+
+[[using-the-loadbalancer]]
+== Using The LoadBalancer
+
+There are load balancers available on
+xref:../install_config/configuring_aws.adoc#[AWS] and
+xref:../install_config/configuring_gce.adoc#[GEC] clouds as well as
+xref:../admin_guide/tcp_ingress_external_ports.adoc#[bare metal].
+
+The new bare-metal
```

Comment:
Let's not say new... we could say (introduced in 3.3) but that's probably not worth it either.